modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/sub.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+export default function sub(string, data) {
+	const REGEX_SUB = /\{\s*([^|}]+?)\s*(?:\|([^}]*))?\s*\}/g;
+
+	if (typeof string !== 'string') {
+		throw new TypeError('the first argument must be a string');
+	}
```

Comment:
This feels a bit redundant to our types file. I don't think it is necessary to add runtime type checking here, especially because it didn't exist here initially.